### PR TITLE
feat: upgrade to webpack 5

### LINKED ISSUE #11: Not compatible with webpack5?
I had to remove webpack-retry-chunk-load-plugin after upgrading to webpack 5.  I was getting errors that 'jsonpScriptSrc' is not defined.

### PR DESCRIPTION
**Pull request by salzhrani**
fixes #11 
based on feedback from [here](https://github.com/webpack/webpack/issues/11816)

I suggest a major version bump

### CODE REVIEW

**Comment by salzhrani** [comment]
did not update testing snapshots, waiting for initial approval of the fix

File: src/index.js
Comment (mattlewis92) [inline src/index.js:32]: Is it ok to use a Map here? I'm just thinking about preserving compatibility for older ES5 browsers, but I know that IE11 does support Map's so maybe it's ok. Do you know if webpack emit's Map objects?

**Review by mattlewis92 — CHANGES_REQUESTED** [review]
This is awesome, thank you so much for taking it on! 😄 Please go ahead and update the test snapshots, I think it's fine to make a major version bump, and I'll make a 1.x branch in case we need any patches there.

File: src/index.js
Comment (salzhrani) [reply src/index.js:32]: I think it is configurable using browserlist

Comment (mattlewis92) [reply src/index.js:32]: Ah ok, let's play it safe then, as the chunk id's should always be strings I think it's ok for these to just be plain old javascript objects. Please can you refactor the 2 Map's just to be objects?

Comment (salzhrani) [reply src/index.js:32]: The loading runtime uses Promises, which is not an ES5 feature. and as per [mdn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map) IE11 supports Map and has, set and get.

Comment (salzhrani) [reply src/index.js:32]: its your call

Comment (mattlewis92) [reply src/index.js:32]: Alright let's keep it as is, if people write in about it breaking people on old browsers it's an easy enough fix

**Review by mattlewis92 — APPROVED** [review]
Thanks again for taking this on! 🙌

**Comment by salzhrani** [comment]
Glad to be of service

**Pull request merged**
